pkg/pipe/pipe.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2021 CGI France
+//
+// This file is part of PIMO.
+//
+// PIMO is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// PIMO is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with PIMO.  If not, see <http://www.gnu.org/licenses/>.
+
+package pipe
+
+import (
+	"github.com/cgi-fr/pimo/pkg/model"
+)
+
+// MaskEngine is a value that always mask the same way
+type MaskEngine struct {
+	pipeline     model.Pipeline
+	injectParent string
+	injectRoot   string
+}
+
+// NewMask return a MaskEngine from a value
+func NewMask(seed int64, injectParent string, injectRoot string, caches map[string]model.Cache, filename string, masking ...model.Masking) (MaskEngine, error) {
+	var definition model.Definition
+	var err error
+	if len(filename) > 0 {
+		definition, err = model.LoadPipelineDefinitionFromYAML(filename)
+		if err != nil {
+			return MaskEngine{nil, injectParent, injectRoot}, err
+		}
+		definition.Seed = seed + 1
```

Comment:
If `pipe` mask is used more than once there is a conflict on the seed. new Seed should be computed with a hash of the relative path.

Reply:
Done with a hash on the path + adding the seed contained in the pipe definition (because user might provide a seed the definition).